api/infra/config/logger.go

```diff
@@ -0,0 +1,53 @@
+package config
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+type Logger struct {
+	debug   *log.Logger
+	info    *log.Logger
+	warning *log.Logger
+	err     *log.Logger
+	writer  io.Writer
+}
+
+func NewLogger(p string) *Logger {
+	writer := io.Writer(os.Stdout)
```

Comment:
writer := io.Writer(os.Stdout) is redundant. You can directly use os.Stdout because it already implements the io.Writer interface. So, you can simplify it to writer := os.Stdout.